### PR TITLE
Faster form interleaved

### PR DESCRIPTION
**Pull request by babrodtk**
This PR improves the performance of `formInterleavedSystem(...)` by assembling the system in a different order. 

It also makes changes the enums for matrix types in `AutoDiffMatrix` to more descriptive names.

### CODE REVIEW

**Comment by bska** [comment]
I haven't looked at this in any kind of details yet, but I wish that 

> changes the enums for matrix types in `AutoDiffMatrix` to more descriptive names.

at least had been a separate commit to ease the review process.

**Comment by babrodtk** [comment]
I agree, my bad.

File: opm/autodiff/NewtonIterationBlackoilInterleaved.cpp [hunk 1]
```diff
@@ -198,52 +198,87 @@ namespace Opm
...
         for (Mat::CreateIterator row = istlA.createbegin(); row != istlA.createend(); ++row) {
-            int ri = row.index();
+            const int ri = row.index();
             for (int i = ia[ri]; i < ia[ri + 1]; ++i) {
                 row.insert(ja[i]);
             }
         }
 
-        const int size = s.rows();
+        // Set all blocks to zero.
```
Comment (bska) [inline opm/autodiff/NewtonIterationBlackoilInterleaved.cpp]: Could you accomplish this using a statement such as
```C++
istlA = 0.0;
```
instead?  The `Dune::BCRSMatrix<>` has an [`operator=(const field_type& k)`](http://www.dune-project.org/doc-2.3.1/doxygen/html/classDune_1_1BCRSMatrix.html#abcb3dea50ab1eac4cb2ab8955af19e82) that appears to do exactly this, albeit in a more straight-line fashion.

**Comment by babrodtk** [comment]
Whitespace fixed.

**Comment by bska** [comment]
> Whitespace fixed.

Thanks.  I'm going to merge this into master now.  Further refinements along the lines of
```C++
istlA = 0.0;
```
can be implemented in the future.

**Pull request merged**